cmd/finch/nerdctl.go

```diff
@@ -142,95 +142,224 @@ func (nc *nerdctlCommand) run(cmdName string, args []string) error {
+					arg = args[i]
+				}
 			}
-		case strings.HasPrefix(arg, "--env-file"):
-			if envArgPos == -1 {
-				envArgPos = i - isDebug
+			// parsing arguments from the command line
+			// may pre-fetch and consume the next argument;
+			// the loop variable will skip any pre-consumed args
+			if skip {
+				skip = false
+				continue
 			}
 
-			shouldSkip, addEnvs, err := handleEnvFile(nc.fs, nc.systemDeps, arg, args[i+1])
-			if err != nil {
-				return err
+			// after last Option position is found, pass through each arg as an internal command argument (short-circuit)
+			if lastOpt {
+				cmdArgs = append(cmdArgs, arg)
+				continue
 			}
-			skip = shouldSkip
-			fileEnvs = append(fileEnvs, addEnvs...)
-		case strings.HasPrefix(arg, "--add-host"):
-			switch arg {
-			case "--add-host":
+
+			switch {
+			case arg == "--debug":
+				nc.logger.SetLevel(flog.Debug)
+			case arg == "--help":
+				nerdctlArgs = append(nerdctlArgs, arg)
+			case arg == "--add-host":
+				// exact match to --add-host
 				args[i+1], err = resolveIP(args[i+1], nc.logger, nc.ecc)
 				if err != nil {
 					return err
 				}
-			default:
+				nerdctlArgs = append(nerdctlArgs, arg)
+			case strings.HasPrefix(arg, "--add-host"):
+				// arg begins with --add-host
 				resolvedIP, err := resolveIP(arg[11:], nc.logger, nc.ecc)
 				if err != nil {
 					return err
 				}
 				arg = fmt.Sprintf("%s%s", arg[0:11], resolvedIP)
+				nerdctlArgs = append(nerdctlArgs, arg)
+			case strings.HasPrefix(arg, "--env-file"):
+				// exact match to --env-file
+				// or arg begins with --env-file
+				shouldSkip, addEnvs, err := handleEnvFile(nc.fs, nc.systemDeps, arg, args[i+1])
+				if err != nil {
+					return err
+				}
+				skip = shouldSkip
+				fileEnvs = append(fileEnvs, addEnvs...)
+			case argIsEnv(arg):
+				// exact match to either -e or --env
+				// or arg begins with -e or --env
+				//     -e="<value>", -e"<value>"
+				//     --env="<key>=<value>", --env"<key>=<value>"
+				shouldSkip, addEnv := handleEnv(nc.systemDeps, arg, args[i+1])
+				skip = shouldSkip
+				if addEnv != "" {
+					envs = append(envs, addEnv)
+				}
+			case shortFlagBoolSet.Has(arg) || longFlagBoolSet.Has(arg):
+				// exact match to: -d, -i, -t
+				// or exact match to: --detach,--init,--interactive,--oom-kill-disable,
+				//                    --privileged,--read-only,--rm,--rootfs,--tty
+				nerdctlArgs = append(nerdctlArgs, arg)
+			case shortFlagBoolSet.Has(arg[:2]):
+				// begins with -d, -i, -t
+				argFlagFinal := false
+				argFlagMid := false
+				lastPos := len(arg) - 1
+				for i, c := range strings.Split(arg, "") {
+					switch {
+					case i == 0:
+						continue
+					case i == lastPos && shortFlagArgsSet.Has("-"+c):
+						argFlagFinal = true
+					default:
+						if shortFlagArgsSet.Has("-" + c) {
+							argFlagMid = true
+						}
+					}
+				}
+				switch {
+				case argFlagMid:
```

Comment:
Can we get rid of this case and handle it in the preceding for loop?

Reply:
- describe my reasoning; acknowledge pertinence
- propose pass odd case through to nerdctl validation

Thanks for the comment. I believe that you are highlighting the possibility of "Error Immediately on Mid Position Detection" and your observation is appropriate for potential optimization.

I similarly considered interrupting immediately, but thought that it may be more clear to have an explicit reaction case. Additionally, I intentionally wanted to "set the runway" if we need to take any additional action to "gracefully" handle the situation.

While we are focusing on this piece of code, I will iterate on this to uniquely print the debug statement, but to ultimately pass this case through to nerdctl arg validation.